src/pepr/operator/controllers/keycloak/authservice/authorization-policy.ts

```diff
@@ -0,0 +1,167 @@
+        {
+          when: [
+            {
+              key: "request.headers[authorization]",
+              notValues: ["*"],
+            },
+          ],
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+function jwtAuthZAuthorizationPolicy(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): AuthorizationPolicy {
+  return {
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: `${name}-jwt-authz`,
+      namespace,
+    },
+    spec: {
+      selector: {
+        matchLabels: labelSelector,
+      },
+      rules: [
+        {
+          from: [
+            {
+              source: {
+                requestPrincipals: [`https://sso.${UDSConfig.domain}/realms/uds/*`],
+              },
+            },
+          ],
+        },
+      ],
+    },
+  };
+}
+
+function authNRequestAuthentication(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): RequestAuthentication {
+  return {
+    kind: "RequestAuthentication",
+    metadata: {
+      name: `${name}-jwt-authn`,
+      namespace,
+    },
+    spec: {
+      jwtRules: [
+        {
+          audiences: [name],
+          forwardOriginalToken: true,
+          issuer: `https://sso.${UDSConfig.domain}/realms/uds`,
+          jwksUri: `http://keycloak-http.keycloak.svc.cluster.local:8080/realms/uds/protocol/openid-connect/certs`,
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+async function updatePolicy(
+  event: AuthServiceEvent,
+  labelSelector: { [key: string]: string },
+  pkg: UDSPackage,
+) {
+  // type safe map event to operation (either Apply or Delete)
+  const operation = operationMap[event.action];
+  const namespace = pkg.metadata!.namespace!;
+  const generation = (pkg.metadata?.generation ?? 0).toString();
+  const ownerReferences = getOwnerRef(pkg);
+
+  const updateMetadata = (resource: AuthorizationPolicy | RequestAuthentication) => {
+    resource!.metadata!.name = resource!.metadata!.name + "-" + generation;
+    resource!.metadata!.ownerReferences = ownerReferences;
+    resource!.metadata!.labels = {
+      "uds/package": pkg.metadata!.name!,
+      "uds/generation": generation,
+    };
+    return resource;
+  };
+
+  try {
+    await K8s(AuthorizationPolicy)[operation](
+      updateMetadata(authserviceAuthorizationPolicy(labelSelector, event.name, namespace)),
+    );
+    await K8s(RequestAuthentication)[operation](
+      updateMetadata(authNRequestAuthentication(labelSelector, event.name, namespace)),
+    );
+    await K8s(AuthorizationPolicy)[operation](
+      updateMetadata(jwtAuthZAuthorizationPolicy(labelSelector, event.name, namespace)),
+    );
+  } catch (e) {
+    log.error(e, `Failed to update auth policy for ${event.name} in ${namespace}: ${e}`);
+    throw new Error(`Failed to update auth policy for ${event.name} in ${namespace}: ${e}`, {
```

Comment:
nit: could pull this common message into `const msg` to use in both spots?

```suggestion
    const msg = `Failed to update auth policy for ${event.name} in ${namespace}: ${e}`
    log.error(e, msg);
    throw new Error(msg, {
```

Reply:
Updated